cmd/migrate/test_helpers.go

```diff
@@ -27,24 +26,195 @@ func RunTransformationTests(t *testing.T, tests []TestCase, transformFunc func([
 		t.Run(tt.Name, func(t *testing.T) {
 			// Parse the input
 			file, diags := hclwrite.ParseConfig([]byte(tt.Config), "test.tf", hcl.InitialPos)
-			assert.False(t, diags.HasErrors())
+			require.False(t, diags.HasErrors(), "Failed to parse input config: %s", diags)
 
 			// Transform the file
 			result, err := transformFunc(file.Bytes(), "test.tf")
-			assert.NoError(t, err)
-			resultString := normalizeWhitespace(string(result))
+			require.NoError(t, err, "Transformation failed")
 
-			// Check each expected output
+			// Check each expected output fragment
+			resultString := string(result)
 			for _, expected := range tt.Expected {
-				normalizedExpected := normalizeWhitespace(expected)
-				formattedResult := ast.FormatString(string(result))
-				formattedExpected := ast.FormatString(expected)
-				assert.Contains(t, resultString, normalizedExpected, cmp.Diff(formattedExpected, formattedResult))
+				assertHCLContains(t, resultString, expected)
 			}
 		})
 	}
 }
 
+// assertHCLContains checks if the expected HCL fragment is present in the actual output
+func assertHCLContains(t *testing.T, actual, expected string) {
+	t.Helper()
+
+	// Trim leading/trailing whitespace from expected as it's usually not significant
+	expected = strings.TrimSpace(expected)
+
+	// First try a smart comparison that handles ordering differences
+	if hclFragmentMatches(actual, expected) {
+		return
+	}
+
+	// If that fails, provide a helpful error message
+	actualFormatted := formatHCLForDiff(actual)
+	expectedFormatted := formatHCLForDiff(expected)
+
+	t.Errorf("Expected HCL fragment not found in output.\n\nExpected fragment:\n%s\n\nActual output:\n%s\n\nDiff:\n%s",
+		expectedFormatted,
+		actualFormatted,
+		cmp.Diff(expectedFormatted, actualFormatted))
+}
+
+// hclFragmentMatches checks if an HCL fragment exists in the actual output
+func hclFragmentMatches(actual, expected string) bool {
+	// First normalize whitespace aggressively for comparison
+	actualNorm := normalizeHCLContent(actual)
+	expectedNorm := normalizeHCLContent(expected)
+
+	// Check if the normalized fragment exists
+	if strings.Contains(actualNorm, expectedNorm) {
+		return true
+	}
+
+	// For more complex cases, try semantic matching
+	return hclSemanticMatch(actual, expected)
+}
+
+// normalizeHCLContent aggressively normalizes HCL for comparison
+func normalizeHCLContent(s string) string {
```

Comment:
there's probably a better way to do this kind of check, but this works for now - can try to simplify later